### ISSUE TITLE
Bookmarks importer fails with sqlite3 error

### ISSUE THREAD

**Issue by murchu27** [qutebrowser/qutebrowser#5913]
I've just installed on Windows 8.1, and I'm trying to import bookmarks from Firefox. I exported from Firefox as HTML, and tried running `python importer.py firefox bookmarks.html`, which resulted in the below error:

```
Traceback (most recent call last):
  File "C:\Users\MMurphy\Syncthing\Downloads\importer.py", line 351, in <module>
    main()
  File "C:\Users\MMurphy\Syncthing\Downloads\importer.py", line 83, in main
    import_function[input_format](args.bookmarks, bookmark_types,
  File "C:\Users\MMurphy\Syncthing\Downloads\importer.py", line 293, in import_moz_places
    places = sqlite3.connect(os.path.join(profile, "places.sqlite"))
sqlite3.OperationalError: unable to open database file
```

Is there something I'm missing here or is this a known issue on Windows?

**Comment by The-Compiler** [comment]
With the import format "mozilla" (which is the default if you pass `firefox` as browser), you will need to pass the path to Firefox' profile folder. To read a HTML export, run the script without a browser given.

I agree this is all a bit confusing though - perhaps we should get rid of the `browser` argument entirely and just take a format (`mozilla`, `netscape`/`html` or `chrome`). @r-c-f @claudehohl any opinion on this?

**Comment by arza-zara** [comment]
Looks like specifying `firefox` makes importer.py look for a firefox profile directory. This should be fixed/clarified but for now you can add `-i netscape`.

**Comment by murchu27** [comment]
Thanks for the response @The-Compiler. The script seems to have run (`STDOUT` got a big dump of lines, one for each bookmark it seems), but I can't see them in qutebrowser now. I'm looking at `qute://bookmarks`, and it still just has the default marks. Any way of troubleshooting this?

**Comment by The-Compiler** [comment]
You will need to redirect its output to qutebrowser's bookmark file - you'll find it in the config folder listed in `:version`. Yet another thing which should probably be improved, or at the very least documented.

**Comment by murchu27** [comment]
That did the trick, thank you! If there's a label for "needs documenting" or something similar, pop it on here and I'll try make some time to take a look at this over the weekend.

**Comment by r-c-f** [comment]
I suspect that the most immediate issue here is the help epilog, which rather unhelpfully suggests that everybody export HTML when doing so these days should almost never be necessary (particularly thanks to the deprecation of IE and the migration of Edge to a Blink core). Leaving the option documentation to explain what is actually required, and the epilog to remind the user of the need to redirect or otherwise add the output to the appropriate qutebrowser config files, would be an immediate improvement I think. Something like
```
usage: importer.py [-h] [-i {netscape,chrome,mozilla}] [-b] [-q] [-s]
                   [--oldconfig] [-B] [-K]
                   [browser] bookmarks

positional arguments:
  browser               Which browser? Omit if using HTML. {chromium, chrome,
                        ie, firefox, seamonkey, palemoon}
  bookmarks             Bookmarks file (html format) or profile folder
                        (Mozilla, Chromium format)

optional arguments:
  -h, --help            show this help message and exit
  -i {netscape,chrome,mozilla}, --input-format {netscape,chrome,mozilla}
                        Which input format? (overrides browser default;
                        "netscape" if neither given)O
  -b, --bookmark-output
                        Output in bookmark format.
  -q, --quickmark-output
                        Output in quickmark format (default).
  -s, --search-output   Output config.py search engine format (negates -B and
                        -K)
  --oldconfig           Output search engine format for old qutebrowser.conf
                        format
  -B, --import-bookmarks
                        Import plain bookmarks (can be combiend with -K)
  -K, --import-keywords
                        Import keywords (can be combined with -B)

To import bookmarks, you'll need the path to your profile or an exported HTML
file from your browser's bookmark manager. Redirect the output from this
script to the appropriate file in your qutebrowser config directory (listed in
the output of :version)
```
for example. This is just what I think I should have done when adding the native browser support; @murchu27 would this have been clearer?.

**Comment by murchu27** [comment]
@r-c-f Yeah that's definitely a bit clearer. On the offchance someone trying to use this isn't overly familiar with the command line, might be helpful to put in an example of what redirection would look like at the very end?

```
To import bookmarks, you'll need the path to your profile or an exported HTML
file from your browser's bookmark manager. Redirect the output from this
script to the appropriate file in your qutebrowser config directory (listed in
the output of :version), for example:

./importer.py firefox ~/.mozilla/firefox/x94s8ry6.default-release > ~/.config/qutebrowser
```

**Comment by The-Compiler** [comment]
Yes, the help epilogue certainly doesn't make things clearer. But I also think there's no point in having both a `browser` and a `--input-format` switch which both mean the same, except in slightly different ways.

I'd instead propose having something like `--input-format {html,chromium-profile,firefox-profile}` and then not take a `browser` argument at all. "html" makes it a bit clearer that this isn't only relevant for Netscape (even if the format was introduced by them), and users of Chromium-based browsers are likely to know that `chromium-profile` is going to work for them as well. We could even point out somewhere that it's usable for other Chromium-based browsers as well.

Agreed on also adding the redirection hint to the help epilogue.

Anyone up for opening a PR for this?

**Comment by The-Compiler** [comment]
Fixed via #5958 - thanks @r-c-f!

**Comment by murchu27** [comment]
Thanks folks!
